components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -152,42 +148,34 @@ const facilityHasUnsavedChanges = () => {
-        || facilityBeforeChange.contact.address.cityEn !== facilitySections.cityEn
-        || facilityBeforeChange.contact.address.addressLine1En !== facilitySections.addressLine1En
-        || facilityBeforeChange.contact.address.addressLine2En !== facilitySections.addressLine2En
-        || facilityBeforeChange.contact.address.prefectureJa !== facilitySections.prefectureJa
-        || facilityBeforeChange.contact.address.cityJa !== facilitySections.cityJa
-        || facilityBeforeChange.contact.address.addressLine1Ja !== facilitySections.addressLine1Ja
-        || facilityBeforeChange.contact.address.addressLine2Ja !== facilitySections.addressLine2Ja
-        || facilityBeforeChange.contact.googleMapsUrl !== facilitySections.googlemapsURL
-        || facilityBeforeChange.mapLatitude.toString() !== facilitySections.mapLatitude
-        || facilityBeforeChange.mapLongitude.toString() !== facilitySections.mapLongitude
-        || JSON.stringify(facilityBeforeChange.healthcareProfessionalIds)
-        !== JSON.stringify(facilitySections.healthcareProfessionalIds)
-        || facilitySections.healthProfessionalsRelations.length
-
-    return areThereUnsavedChanges
+    const facilitySections = facilitiesStore.facilitySectionFields
+
+    return facilityBeforeChange.nameEn !== facilitySections.nameEn
+      || facilityBeforeChange.nameJa !== facilitySections.nameJa
+      || facilityBeforeChange.contact.phone !== facilitySections.phone
+      || facilityBeforeChange.contact.website !== facilitySections.website
+      || facilityBeforeChange.contact.email !== facilitySections.email
+      || facilityBeforeChange.contact.address.postalCode !== facilitySections.postalCode
+      || facilityBeforeChange.contact.address.prefectureEn !== facilitySections.prefectureEn
+      || facilityBeforeChange.contact.address.cityEn !== facilitySections.cityEn
+      || facilityBeforeChange.contact.address.addressLine1En !== facilitySections.addressLine1En
+      || facilityBeforeChange.contact.address.addressLine2En !== facilitySections.addressLine2En
+      || facilityBeforeChange.contact.address.prefectureJa !== facilitySections.prefectureJa
+      || facilityBeforeChange.contact.address.cityJa !== facilitySections.cityJa
+      || facilityBeforeChange.contact.address.addressLine1Ja !== facilitySections.addressLine1Ja
+      || facilityBeforeChange.contact.address.addressLine2Ja !== facilitySections.addressLine2Ja
+      || facilityBeforeChange.contact.googleMapsUrl !== facilitySections.googlemapsURL
+      || facilityBeforeChange.mapLatitude.toString() !== facilitySections.mapLatitude
+      || facilityBeforeChange.mapLongitude.toString() !== facilitySections.mapLongitude
+      || JSON.stringify(facilityBeforeChange.healthcareProfessionalIds)
+      !== JSON.stringify(facilitySections.healthcareProfessionalIds)
+      || facilitySections.healthProfessionalsRelations.length > 0
```

Comment:
🍷 The reason for adding the ` > 0` here, is because the return type was being interpreted as `boolean | number` when we definitely don't want a number.